src/js/AgeVerification.tsx

```diff
@@ -0,0 +1,210 @@
+language_file.translations = HTML.escape(language_file.translations);
+
+import Languages from "./components/Languages";
+import Footer from "./components/Footer";
+
+await import((`../css/AgeVerification.pcss`));
+
+function parseContent(str: string, context = undefined) {
+  return HTML.parseMarkdown(HTML.escape(context ? parseContext(str, context) : str));
+}
+
+function verifyInputs(options: VerificationOptions) {
+  if (!options.branding?.name) throw new Error("Company name is required");
+  else if (!options.branding?.logo) throw new Error("Company logo is required");
+  else if (!options.branding?.links?.cookies) throw new Error("Link to company cookie policy is required");
+  else if (!options.branding?.links?.termsOfService) throw new Error("Link to company terms of service is required");
+  else if (!isValidURL(options.branding.logo)) throw new Error("Invalid logo URL provided");
+  else if (!isValidURL(options.branding.links.cookies)) throw new Error("Invalid cookies URL provided");
+  else if (!isValidURL(options.branding.links.termsOfService)) throw new Error("Invalid terms of service URL provided");
+}
+
+/* 1) Use first language in file as default
+ * 2) Use website's primary language
+ * 3) Use client's primary language if available */
+function initializeLanguage(options: VerificationOptions) {
+  let currentLanguage = Object.values(language_file.translations)[0];
+  if (language_file.translations[options.primary_language]) currentLanguage = language_file.translations[options.primary_language];
+  const userLang = ((navigator.language) ? navigator.language : navigator['userLanguage']).split("-")[0].toUpperCase();
+  if (language_file.translations[userLang]) currentLanguage = language_file.translations[userLang];
+
+  return currentLanguage;
+}
+
+function shouldShowModal(options: VerificationOptions) {
+  const cookieUrl = getAbsoluteURL(options.branding.links.cookies, true) as URL,
+    termsOfServiceUrl = getAbsoluteURL(options.branding.links.termsOfService, true) as URL;
+
+  // return immediately if the user is on cookie or TOS page
+  if (cookieUrl.host === window.location.host && cookieUrl.pathname === window.location.pathname) return false;
+  else if (termsOfServiceUrl.host === window.location.host && termsOfServiceUrl.pathname === window.location.pathname) return false;
+
+  return true;
+}
+
+const AgeVerification = {
+  /**
+   * @returns {Promise<boolean>} True whether the user is of age and agrees to TOS and cookie policy, false otherwise
+   */
+  prompt: (options?: VerificationOptions): Promise<boolean> => {
+    options = Object.assign({
+      minimumAge: 18,
+      cookie_ttl: 2592000,
+      disableScrolling: true,
+      backgroundBlur: true,
+      models: ["kERQDXH", "8aLOs5g", "atUVnRj", "hBn9ona", "XDszZnM", "lHojdWj", "eTQTZtz", "C6Xx8rH", "j2odobq", "5xsHsyo", "omyy7Ct", "FYFCgJr", "Bbww6Ud", "RJ8FNDb", "gKhQO5E", "Vz7yaEy", "1JsqDGw", "S29mzqv"].map(e => `https://i.imgur.com/${e}.jpg`),
```

Comment:
No imgur pls. They have their own terms of service and stuff, and the users of this library shouldn't have to worry about this